pkg/scheduler/scheduler.go

```diff
@@ -612,10 +612,15 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 			if klog.V(2).Enabled() {
 				klog.InfoS("Successfully bound pod to node", "pod", klog.KObj(pod), "node", scheduleResult.SuggestedHost, "evaluatedNodes", scheduleResult.EvaluatedNodes, "feasibleNodes", scheduleResult.FeasibleNodes)
 			}
-
 			metrics.PodScheduled(prof.Name, metrics.SinceInSeconds(start))
 			metrics.PodSchedulingAttempts.Observe(float64(podInfo.Attempts))
-			metrics.PodSchedulingDuration.Observe(metrics.SinceInSeconds(podInfo.InitialAttemptTimestamp))
+
+			// We breakdown the pod scheduling duration by attempts (capped to a limit).
+			attempts := podInfo.Attempts
+			if attempts > 50 {
+				attempts = 50
+			}
+			metrics.PodSchedulingDuration.WithLabelValues(string(attempts)).Observe(metrics.SinceInSeconds(podInfo.InitialAttemptTimestamp))
```

Comment:
Could you send a parent PR that changes the variable name `InitialAttemptTimestamp` to `QueueuingTimestamp` or something like that?

Reply:
We should do that in a follow up PR to allow for patching this PR back if needed.

Reply:
How about explicitly expressing that internally we have already "bucked" the attempts (we can adjust the buckets like [5,10), [10, 20), [20, +inf): 

```go
metrics.PodSchedulingDuration.WithLabelValues(displayAttempts(podInfo.Attempts)).Observe(metrics.SinceInSeconds(podInfo.InitialAttemptTimestamp))

func displayAttempts(n int) string {
	if n < 5 {
		return string(n)
	} else if n < 20 {
		return "[5, 20)"
	}
	return "[20, +inf)"
}
```

Reply:
The most important information from this is actually distinguishing pods scheduled from the first attempt, so perhaps we can change the label to first_attempt, and the value is true or false.

Reply:
I'm inclined to the current version as it exposes more info (maybe not that useful for now), while also able to infer whether it's the first attempt or not.

Reply:
sg, I increased the limit to 15, I think it is small enough cardinality, and will likely cover most cases. We should increase the limit for PodSchedulingAttempts, I think 5 is too low.

Reply:
Shouldn't we display 15 as `[15, +inf)` or something similar? so that users would know it's not literally 15.

Reply:
changed it to "15+", likely clear enough, sg?

Reply:
SG.